docs/plugin_development.md

```diff
@@ -0,0 +1,161 @@
+# Developing a Seabird Plugin
+
+Seabird plugins require a bit of boilerplate, but should be reasonably simple to create.
+
+Unless you're creating a [URL plugin](./url_plugin_development.md), you should put your plugins under `plugins/extra/`.
+
+## Defining a New Plugin
+
+Here's a simple plugin to get you started:
+
+```go
+package extra
+
+import (
+    seabird "github.com/belak/go-seabird"
+)
+
+func init() {
+    seabird.RegisterPlugin("my_cool_plugin", newMyCoolPlugin)
+}
+
+func newMyCoolPlugin(b *seabird.Bot) error {
+    cm := b.CommandMux()
+
+    cm.Event("my_command", commandCallback, &seabird.HelpInfo{
+        Description: "This command does something.",
+    })
+
+    return nil
+}
+
+func commandCallback(r *seabird.Request) {
+    r.MentionReply("You ran my_command!")
+}
+```
+
+This plugin adds a single command called `my_command` that will reply to a user when the command is called.
+
+Note that you must add the plugin's name (`my_cool_plugin` from `init()`) to your bot's configuration under the `plugins` list for the new command to be active.
+
+## Registering Callbacks
+
+You've already seen one way to register bot callbacks in `CommandMux.Event`. There are a few different ways to register message callbacks in a plugin:
+
+### `BasicMux`
+
+`BasicMux().Event`: This will register a callback that will be called when Seabird sees specific raw IRC commands like `JOIN`, `PART`, and `KICK`.
```

Comment:
Nit: I'd probably do something like `BasicMux{}.` as that's what denotes a type, rather than a function call.

That being said, do we want to document things here as well as in the godoc?

Reply:
I put them here so that a developer can see them all in one place, instead of looking into `mux_basic.go` and `mux_command.go`, but if it'd be better just linking to godoc that's fine as long as the choices are shown and the differences are clear. It's definitely a bummer to duplicate.

Reply:
https://godoc.org/github.com/belak/go-seabird

We could update this with examples and such... there's also https://godoc.org/github.com/fluhus/godoc-tricks which documents a number of ways of including useful information in the godoc